File: src/main/java/com/freya02/botcommands/application/slash/SlashCommandParameter.java
package com.freya02.botcommands.application.slash;

import com.freya02.botcommands.application.ApplicationCommandParameter;
import com.freya02.botcommands.parameters.SlashParameterResolver;

import java.lang.reflect.Parameter;

public class SlashCommandParameter extends ApplicationCommandParameter<SlashParameterResolver> {
	public SlashCommandParameter(Parameter parameter, int index) {
		super(SlashParameterResolver.class, parameter, index);
	}
}

File: src/main/kotlin/io/github/freya022/botcommands/api/commands/application/slash/SlashCommandParameter.kt
package io.github.freya022.botcommands.api.commands.application.slash

import io.github.freya022.botcommands.api.commands.application.ApplicationCommandParameter

/**
 * Represents a slash command parameter,
 * which has its value computed by an aggregation function, from one or more options.
 */
interface SlashCommandParameter : ApplicationCommandParameter {
    override val nestedAggregatedParameters: List<SlashCommandParameter>

    override val command: SlashCommandInfo
}